src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,850 @@
+        u_avg[3]/(g*u_avg[1]*drho);;
+        u_avg[3]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho)];
+      [1.0/(g*drho);;
+        u_avg[2]/(g*u_avg[1]*drho);;
+        u_avg[3]/(g*u_avg[1]*drho);;
+      -1.0/(g*drho);;
+      -u_avg[5]/(g*u_avg[4]*drho);;
+      -u_avg[6]/(g*u_avg[4]*drho)];
+      [u_avg[5]/(g*u_avg[4]*drho);;
+        u_avg[2]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+      -u_avg[5]/(g*u_avg[4]*drho);;
+        (g*rho1*u_avg[4]^3 - g*rho2*u_avg[4]^3 - rho2*u_avg[5]^2)/(g*rho2*u_avg[4]^2*drho);;
+      -u_avg[5]*u_avg[6]/(g*u_avg[4]^2*drho)];
+      [u_avg[6]/(g*u_avg[4]*drho);;
+        u_avg[2]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho);;
+      -u_avg[6]/(g*u_avg[4]*drho);;
+      -u_avg[5]*u_avg[6]/(g*u_avg[4]^2*drho);;
+        (g*rho1*u_avg[4]^3 - g*rho2*u_avg[4]^3 - rho2*u_avg[6]^2)/(g*rho2*u_avg[4]^2*drho)]]
+
+  # Add dissipation to entropy conservative flux to obtain entropy stable flux
+  f_es = f_ec - 0.5 * λ * H * (q_rr - q_ll)
+  
+  return SVector(f_es[1], f_es[2], f_es[3], f_es[4], f_es[5], f_es[6], zero(eltype(u_ll)))
+end
+
+
+# Calculate maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed
+@inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer, 
+  equations::TwoLayerShallowWaterEquations2D)
+
+  v1_ll, w1_ll, v2_ll, w2_ll = velocity(u_ll, equations)
+  v1_rr, w1_rr, v2_rr, w2_rr = velocity(u_rr, equations)
+
+  if orientation == 1
+    Um_ll = (u_ll[2] + u_ll[5]) / (u_ll[1] + u_ll[4])
+    Um_rr = (u_rr[2] + u_rr[5]) / (u_rr[1] + u_rr[4])
+  else
+    Um_ll = (u_ll[3] + u_ll[6]) / (u_ll[1] + u_ll[4])
+    Um_rr = (u_rr[3] + u_rr[6]) / (u_rr[1] + u_rr[4])
+  end
+
+  # Calculate the wave celerity on the left and right
+  h1_ll, h2_ll = waterheight(u_ll, equations)
+  h1_rr, h2_rr = waterheight(u_rr, equations)
+
+  c_ll = sqrt(equations.gravity * (h1_ll + h2_ll) )
+  c_rr = sqrt(equations.gravity * (h1_rr + h2_rr))
+
+  return (max(abs(Um_ll),abs(Um_rr)) + max(c_ll,c_rr))
+end
+
+
+@inline function max_abs_speed_naive(u_ll, u_rr, 
+                                     normal_direction::AbstractVector,
+                                     equations::TwoLayerShallowWaterEquations2D)
+  # Extract and compute the velocities in the normal direction
+  v1_ll, w1_ll, v2_ll, w2_ll = velocity(u_ll, equations)
+  v1_rr, w1_rr, v2_rr, w2_rr = velocity(u_rr, equations)
+
+  vel1_ll = v1_ll * normal_direction[1] + w1_ll * normal_direction[2]
+  vel1_rr = v1_rr * normal_direction[1] + w1_rr * normal_direction[2]
+  vel2_ll = v2_ll * normal_direction[1] + w2_ll * normal_direction[2]
+  vel2_rr = v2_rr * normal_direction[1] + w2_rr * normal_direction[2]
+  
+  Um_ll = (vel1_ll+vel2_ll)/(u_ll[1]+u_ll[4])
+  Um_rr = (vel1_rr+vel2_rr)/(u_rr[1]+u_rr[4])
```

Comment:
```suggestion
  Um_ll = (vel1_ll + vel2_ll) / (u_ll[1] + u_ll[4])
  Um_rr = (vel1_rr + vel2_rr) / (u_rr[1] + u_rr[4])
```

Is there a reference to motivate this choice of the velocity? Why is it divided by the water heights again?

Reply:
Good that you ask, this is actually a bug I did not notice before. 
`Um` is something like a bulk velocity we get from the approximation of the eigenvalues, as is used in most two-layer papers. A good description is given in a paper from Nycander (doi:10.1016/j.ocemod.2008.06.003).

The correct formula is:
`Um_ll = (vel1_ll * u_ll[1] + vel2_ll * u_ll[4]) / (u_ll[1] + u_ll[4])` 
`Um_rr = (vel1_rr * u_rr[1] + vel2_rr * u_rr[4]) / (u_rr[1] + u_rr[4])`

Reply:
I fixed the formula and added a description in commit d35f75d5a06074c87e9f5a5f04a8b2946d18c110